### DISCUSSION TITLE
Converting THIS raw to a webp.

### DISCUSSION THREAD

**Discussion by mhoppebeyonity** [libvips/libvips#4228] [Help wanted]
Hello nice people. Thank you for reading.

I have a file that is made from from **raw** pixel **bgra**. Size is 2880x1620.

I can convert that into a webp with imagemagick, but i would love to do that with libvips. I am on windows and have the latest binaries. so could use vips.exe.

But i cannot find the right cli-parameters. Could anyone help me wioth that, please?
[render0.zip](https://github.com/user-attachments/files/17558608/render0.zip)

this is the command line that works with imagemagick:

start /wait magick convert -size 2880x1620 -depth 8 -quality 80 bgra:a:\render0.pixel render0.webp

**Comment by jcupitt** [comment]
Hello @mhoppebeyonity,

You can use:

```
$ vips rawload render0.pixel x.webp 2880 1620 4
```

Though the band order is wrong, you'll need to swap B and R. I use python for things like this, eg.:

```python
image = pyvips.Image.rawload("render0.pixel", 2880, 1620, 4)
image = image[2].bandjoin([image[1], image[0])
```

**Reply by mhoppebeyonity** [reply]
Wow! Thank you so much!

Actually I might be able to change the order of the channels when saving the pixel file. Since the conversion is very time critical I'd like to avoid python or multiple image manipulation steps.

However, do you see any changes to make vips load the image with channels swept, so it not necessarily must be done in an extra step? So that I could use the command line executable in one go? 

Thank you so much!

**Reply by jcupitt** [reply]
Sorry -- the libvips CLI is deliberately very simple, you can only run one operation at once. You're supposed to use some kind of scripting language for anything complex.
